x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/BaseEnrollmentTokenGenerator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.enrollment;
+
+import org.elasticsearch.common.ssl.SslKeyConfig;
+import org.elasticsearch.common.ssl.SslUtil;
+import org.elasticsearch.common.ssl.StoreKeyConfig;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.xpack.core.ssl.SSLService;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.URI;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BaseEnrollmentTokenGenerator {
+
+    public BaseEnrollmentTokenGenerator() {
```

Comment:
I'm not particularly tied to this abstraction model but we have two use cases for enrollment token generation:

1. The one from the CLI tool that runs outside the node and needs to use CommanLineHttpClient
2. The one on node startup that runs inside the node

It doesn't make sense for the second to do HTTP calls to the node and the first can't use the node client. They both share a couple of functionality parts like calculating the ca fingerprint or the logic to filter and prioritize IP addresses.

My attempt here is to reuse the common functionality while allowing each type of enrollment token generation to perform its actions based on specific needs and constraints. I'm open to other suggestions